File: tests/regression/r_static/util/convert_features_and_add_preproc.R
convert_features_and_add_preproc <- function(prefix) {
	if (!exists(paste(prefix, 'order', sep=''), mode='numeric')) {
		return(FALSE)
	}

	ftype <- eval(parse(text=paste(prefix, 'feature_type', sep='')))
	if (regexpr('Ulong', ftype)>0) {
		type <- 'ULONG'
	} else if (regexpr('Word', ftype)>0) {
		type <- 'WORD'
	} else {
		return(FALSE)
	}

	order <- eval(parse(text=paste(prefix, 'order', sep='')))
	gap <- eval(parse(text=paste(prefix, 'gap', sep='')))
	reverse <- eval(parse(text=paste(prefix, 'reverse', sep='')))
	sg('add_preproc', paste('SORT', type, 'STRING', sep=''))
	sg('convert', 'TRAIN', 'STRING', 'CHAR', 'STRING', type,
		order, order-1, gap, reverse)
	sg('attach_preproc', 'TRAIN')

	sg('convert', 'TEST', 'STRING', 'CHAR', 'STRING', type,
		order, order-1, gap, reverse)
	sg('attach_preproc', 'TEST')

	return(TRUE)
}

File: tests/regression/matlab_and_octave/util/convert_features_and_add_preproc.m
function y = convert_features_and_add_preproc(prefix)
	eval(sprintf(['global ', prefix, 'order']));
	eval(sprintf(['global ', prefix, 'gap']));
	eval(sprintf(['global ', prefix, 'reverse']));
	eval(sprintf(['global ', prefix, 'feature_type']));
	order=eval(sprintf([prefix, 'order']));
	gap=eval(sprintf([prefix, 'gap']));
	reverse=eval(sprintf([prefix, 'reverse']));
	feature_type=eval(sprintf([prefix, 'feature_type']));

	y=false;

	if isempty(order)
		return;
	end

	if strcmp(feature_type, 'Ulong')==1
		type='ULONG';
	elseif strcmp(feature_type, 'Word')==1
		type='WORD';
	else
		return;
	end

	sg('add_preproc', strcat('SORT', type, 'STRING'));
	sg('convert', 'TRAIN', 'STRING', 'CHAR', 'STRING', type, ...
		order, order-1, gap, reverse);
	sg('attach_preproc', 'TRAIN');

	sg('convert', 'TEST', 'STRING', 'CHAR', 'STRING', type, ...
		order, order-1, gap, reverse);
	sg('attach_preproc', 'TEST');

	y=true;
